Execution environment image should not be required

willthames opened the pull request:
##### SUMMARY
Remove requirement from image parameter as it's not always needed (e.g. when state is absent or updating an existing execution environment)

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - Collection
##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 22.1.1.dev353+g6c4c8df3f8
`

AlanCoding commented on awx_collection/plugins/modules/execution_environment.py:
```diff
@@ -79,17 +78,21 @@ def main():
+    required_if = [
+        ['state', 'present', ['image']],
+        ['state', 'exists', ['image']],
+    ]
```
I would prefer to not even have this condition. Someone could modify a different field of an existing item. If it is missing (on creation) then the API error should suffice.

willthames committed:
Execution environment image should not be required

Remove requirement from image parameter as it's not always needed
(e.g. when state is absent or updating an existing execution
environment)

willthames commented:
@AlanCoding I've updated the PR based on your feedback - image is now just not required.

AlanCoding commented:
Thanks. This looks good to me.

Could you do the DCO signoff before we merge?

https://github.com/ansible/awx/blob/159dd62d84df3ac14d274571724eee0324c2ca61/DCO_1_1.md?plain=1#L4
